src/StelLogger.cpp

```diff
@@ -44,193 +78,352 @@ void StelLogger::init(const QString& logFilePath)
-				writeLog(line);
+			if (line.startsWith("processor", Qt::CaseInsensitive))
+				ncpu++;
+
+			if (line.startsWith("model name", Qt::CaseInsensitive) && readModel)
+			{
+				cpumodel = line.split(":").last().trimmed();
+				readModel = false;
+			}
+
+			// for ARM-devices, such Raspberry Pi
+			if (line.startsWith("hardware", Qt::CaseInsensitive))
+				hardware = line.split(":").last().trimmed();
+			if (line.startsWith("model", Qt::CaseInsensitive) && !hardware.isEmpty())
+				model = line.split(":").last().trimmed();
 		}
 		infoFile.close();
 	}
 
-	QProcess lspci;
-	lspci.start("lspci", { "-v" }, QIODevice::ReadOnly);
-	lspci.waitForFinished(300);
-	const QString pciData(lspci.readAll());
-	#if (QT_VERSION>=QT_VERSION_CHECK(5, 14, 0))
-	QStringList pciLines = pciData.split('\n', Qt::SkipEmptyParts);
-	#else
-	QStringList pciLines = pciData.split('\n', QString::SkipEmptyParts);
-	#endif
-	for (int i = 0; i<pciLines.size(); i++)
+	QString freq = "unknown";
+	infoFile.setFileName("/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq");
+	if (infoFile.open(QIODevice::ReadOnly | QIODevice::Text))
 	{
-		if(pciLines.at(i).contains("VGA compatible controller"))
-		{
-			writeLog(pciLines.at(i));
-			i++;
-			while(i < pciLines.size() && pciLines.at(i).startsWith('\t'))
-			{
-				if(pciLines.at(i).contains("Kernel driver in use"))
-					writeLog(pciLines.at(i).trimmed());
-				else if(pciLines.at(i).contains("Kernel modules"))
-					writeLog(pciLines.at(i).trimmed());
-				i++;
-			}
-		}
+		// frequency in kHz: https://www.kernel.org/doc/Documentation/cpu-freq/user-guide.txt
+		freq = QString::number( infoFile.readAll().toInt()/1000);
+		infoFile.close();
 	}
-#endif
 
-// Aargh Windows API
-#elif defined Q_OS_WIN
-	MEMORYSTATUSEX statex;
-	statex.dwLength = sizeof (statex);
-	GlobalMemoryStatusEx(&statex);
-	writeLog(QString("Total physical memory: %1 MB").arg(statex.ullTotalPhys/(1024<<10)));
-	writeLog(QString("Available physical memory: %1 MB").arg(statex.ullAvailPhys/(1024<<10)));
-	writeLog(QString("Physical memory in use: %1%").arg(statex.dwMemoryLoad));
-	#ifndef _WIN64
-	// This always reports about 8TB on Win64, not really useful to show.
-	writeLog(QString("Total virtual memory: %1 MB").arg(statex.ullTotalVirtual/(1024<<10)));
-	writeLog(QString("Available virtual memory: %1 MB").arg(statex.ullAvailVirtual/(1024<<10)));
-	#endif
+	if (cpuOK)
+	{
+		writeLog(QString("Processor name: %1").arg(cpumodel));
+		if (!hardware.isEmpty())
+			writeLog(QString("Processor hardware: %1").arg(hardware));
+		if (!model.isEmpty())
+			writeLog(QString("Device model: %1").arg(model));
+		writeLog(QString("Processor maximum speed: %1 MHz").arg(freq));
+		writeLog(QString("Processor logical cores: %1").arg(ncpu));
+	}
 
-	HKEY hKey = Q_NULLPTR;
-	DWORD dwType = REG_DWORD;
-	DWORD numVal = 0;
-	DWORD dwSize = sizeof(numVal);
+	// memory info
+	struct sysinfo memInfo;
+	sysinfo (&memInfo);
+	long long totalRAM = memInfo.totalram;
```

Comment:
This should be unsigned. And maybe better to use a sized typedef, like `uint64_t`.